internal/cli/project_apply.go

```diff
@@ -0,0 +1,382 @@
+package cli
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/posener/complete"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	configpkg "github.com/hashicorp/waypoint/internal/config"
+	"github.com/hashicorp/waypoint/internal/datasource"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+type ProjectApplyCommand struct {
+	*baseCommand
+
+	flagDataSource     string
+	flagGitURL         string
+	flagGitRef         string
+	flagGitAuthType    string
+	flagGitUsername    string
+	flagGitPassword    string
+	flagGitKeyPath     string
+	flagGitKeyPassword string
+	flagWaypointHcl    string
+}
+
+func (c *ProjectApplyCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	flagSet := c.Flags()
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+	args = flagSet.Args()
+	ctx := c.Ctx
+
+	if len(args) != 1 {
+		c.ui.Output("Single argument required.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	name := args[0]
+
+	sg := c.ui.StepGroup()
+	defer sg.Wait()
+
+	s := sg.Add("Checking for an existing project named: %s", name)
+	defer func() { s.Abort() }()
+
+	// Check for an existing project of the same name.
+	resp, err := c.project.Client().GetProject(ctx, &pb.GetProjectRequest{
+		Project: &pb.Ref_Project{
+			Project: name,
+		},
+	})
+	if status.Code(err) == codes.NotFound {
+		// If the error is a not found error, act as though there is no error
+		// and the project is nil so that we can handle that later.
+		resp = nil
+		err = nil
+	}
+	if err != nil {
+		c.ui.Output(
+			"Error checking for project: %s", clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	// Setup our project that we're going to override
+	var proj *pb.Project
+	if resp != nil {
+		s.Update("Updating project %q...", name)
+		proj = resp.Project
```

Comment:
For use in step confirmation
```suggestion
	var updated bool
	if resp != nil {
		s.Update("Updating project %q...", name)
		updated = true
```